### PR TITLE
adapt for upcoming changes in PyPy's SOABI tag

### LINKED ISSUE #372: Transition PyPy to PEP 3149 compliant SOABI tags
It turns out PyPy's SOABI tag is not compliant with PEP 3149. Instead of containing both the ABI tag and the platform tag, PyPY's value returned for the SOABI query is only the ABI tag. For instance, on linux 64 CPython returns `cpython-38-x86_64-linux-gnu` but PyPy returns `pypy36-pp73` where the correct value is `pypy36-pp73-x86_64-linux-gnu`.

I want to fix this on PyPy for our upcoming release, but then [`get_abi_tag`](https://github.com/pypa/wheel/blob/0.35.1/src/wheel/bdist_wheel.py#L67) will break. It correctly trims the platform tag off the SOABI value on line 94. Eventually we should get to a situation where line 94 works for both PyPy and CPython, but there needs to be a transition period based on either direct value parsing or some kind of version check.

Any thoughts? PyPy users are pretty good about updating, I think within a year we could drop any temporary shims we put in to work around this issue, under the assumption that by then anyone building a PyPy wheel will have upgraded to the latest version.

### PR DESCRIPTION
**Pull request by mattip**
Closes #372. PyPy currently has a SOABI tag like `pypy36-pp73` (only the ABI tag) but according to PEP 3149 it should be `pypy36-pp73-x86_64-linux-gnu` (ABI tag - platform tag). PyPy may change this in an upcoming release.

This should be forward compatible, as long as the ABI tag contains a single `-`.

### CODE REVIEW

**Review by auvipy — APPROVED** [review]

**Comment by agronholm** [comment]
One more thing – could you add a news entry for this? I'm not sure I understand these changes well enough.

**Comment by mattip** [comment]
added a news blurb, feel free to fix it up. It seems too wordy to me.

**Comment by agronholm** [comment]
Sorry for not being clear – I was hoping for something more concise for the news item, maybe a one-liner. I can think of something on my own too.

**Comment by mattip** [comment]
redone now that I reviewed what I did and understood it more completely

**Comment by mattip** [comment]
maybe "'Future-proof calculating the ABI tag from PyPy's SOABI tag" ?

**Comment by agronholm** [comment]
Thanks!

**Comment by mattip** [comment]
Thank you!
